internal/tiltfile/tiltextension/plugin_test.go

```diff
@@ -191,6 +195,311 @@ include('Tiltfile.b')
+v1alpha1.extension(name='my-ext', repo_name='custom')
+v1alpha1.extension(name='my-ext-with-path', repo_name='custom', repo_path='subdir2')
+
+# Assert that loading an extension without a repo_path loads from the repo-wide path
+load("ext://my-ext", "printExt")
+printExt()
+
+load("ext://my-ext-with-path", "printExt2")
+printExt2()
+`, f.tmp.Path()))
+
+	extContent := `
+def printExt():
+	print("main ext")
+	`
+
+	extContent2 := `
+def printExt2():
+	print("sub ext")
+	`
+
+	f.tmp.WriteFile(filepath.Join("ext-repo", "subdir", "Tiltfile"), extContent)
+	f.tmp.WriteFile(filepath.Join("ext-repo", "subdir", "subdir2", "Tiltfile"), extContent2)
+
+	res := f.assertExecOutput("main ext\nsub ext")
+	f.assertLoadRecorded(res, "my-ext", "my-ext-with-path")
+}
+
+func TestRepoLoadHostAndSubpath(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		// We don't want to have to bother with file:// escaping on windows.
+		// The repo reconciler already tests this.
+		t.Skip()
+	}
+
+	t.Skip("cannot use file:// repos with git_subpath, making this test unusable")
+
+	// Assert that extension repositories with a defined subpath load registered extensions
+	// from that subpath, including autoregistration by host match
+	f := newExtensionFixture(t)
+
+	f.tiltfile(fmt.Sprintf(`
+v1alpha1.extension_repo(name='custom', url='file://%s/ext-repo', load_host='custom', git_subpath='subdir')
+
+# Should load an extension from the custom repo at <repo.path>/my-ext
+load("ext://custom/my-ext", "printExt")
+printExt()
+
+# Should load from <repo.path>/my-ext/subext
+load("ext://custom/my-ext/subext", "printSub")
+printSub()
+`, f.tmp.Path()))
+
+	extContent := `
+def printExt():
+	print("main ext")
+	`
+
+	subExtContent := `
+def printSub():
+	print("sub ext")
+	`
+
+	f.tmp.WriteFile(filepath.Join("ext-repo", "subdir", "my-ext", "Tiltfile"), extContent)
+	f.tmp.WriteFile(filepath.Join("ext-repo", "subdir", "my-ext", "subext", "Tiltfile"), subExtContent)
+
+	res := f.assertExecOutput("main ext\nsub ext")
+	f.assertLoadRecorded(res, "custom/my-ext", "custom/my-ext/subext")
+}
+
+// Verifies behavior around registering an extension using the default repository as a fallback
+func TestRegisterDefaultExtension(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		// We don't want to have to bother with file:// escaping on windows.
+		// The repo reconciler already tests this.
+		t.Skip()
+	}
+
+	f := newExtensionFixture(t)
+
+	p := NewFakePlugin(f.extrr, f.extr)
+
+	// NOTE: I don't know a better way to get a model, and this test needs one to get an object set
+	f.tiltfile(`print("hello")`)
+	model, _ := f.skf.ExecFile("Tiltfile")
+	objSet, err := tiltfilev1alpha1.GetState(model)
+	if err != nil {
+		f.t.Fatalf("unexpected error %v", err)
+	}
+
+	moduleName := "tests/golang"
+	extName := apis.SanitizeName(moduleName)
+	extSet := objSet.GetOrCreateTypedSet(&v1alpha1.Extension{})
+
+	ext := p.registerDefaultExtension(nil /* *starlark.Thread */, extSet, extName, moduleName)
```

Comment:
All of these tests of the extension loading helpers are a bit crufty since I'm not sure how else to get a handle on a `model`, which I need for an `objSet`. I'm also not sure how to get a `*starlark.Thread`, but these functions don't actually *use* the thread so I pass in `nil`.